src/interface/sparse_pack.hpp

```diff
@@ -437,8 +340,71 @@ class SparsePack : public SparsePackBase {
     const int vidx = GetLowerBound(b, t) + t.idx;
     return pack_(dir, b, vidx)(k, j, i);
   }
+
+  template <class... VTs>
+  auto GetPtrs(const int b, const TE el, int k, int j, int i, VTs... vts) {
+    return std::make_tuple({&(*this)(b, el, vts, k, j, i)...});
+  }
 };
 
+inline auto MakePackDescriptor(StateDescriptor *psd, const std::vector<std::string> &vars,
```

Comment:
What is the implication of passing in the state descriptor here? Obviously, that's needed to get at the UIDs... but should we warn people somewhere to only use the `resolved_pacakges` state descriptor? Or alternatively provide an overload that takes a pointer to the mesh so we can unambiguously get the right one?

Reply:
My thinking was that a `StateDesciptor` is actually where you want to go for all of the field definitions (since in some cases `MeshBlockData` doesn't have all of the fields present). It may be that you want to create a pack descriptor that only gets fields from a certain package as well. That being said, I agree it would be good to have an overloaded `MakePackDesciptor` that takes `MeshData` or `MeshBlockData` and gets the `resolved_packages` state descriptor.